lib/atrium/paginate.rb

```diff
@@ -0,0 +1,56 @@
+require "URI"
+
+module Atrium
+  module Paginate
+    DEFAULT_PAGE = 1
+    DEFAULT_RECORDS_PER_PAGE = 25
+    DEFAULT_TOTAL_PAGES = 1
+
+    attr_accessor :current_page, :endpoint, :klass_name, :per_page, :total_pages
+
+    def endpoint_name(query_params = nil)
+      set_klass_name
+
+      if query_params.present?
+        @endpoint = klass_name + "?" + URI.encode_www_form(query_params) + "&"
+      else
+        @endpoint = klass_name + "?"
+      end
+    end
+
+    def paginate_endpoint(query_params = nil)
+      endpoint_name(query_params)
+      set_pagination_fields
+      response_list
+    end
+
+    def response_list
+      list = []
+
+      until current_page > total_pages
+        paginated_endpoint =  endpoint + "page=#{current_page}&records_per_page=#{per_page}"
+        response = ::Atrium.client.make_request(:get, paginated_endpoint)
+
+        # Add new objects to the list
+        response["#{klass_name}"].map do |params|
+          list << self.new(params)
+        end
+        @current_page += 1
+      end
+      list
+    end
+
+    def set_klass_name
```

Comment:
why doesn't we just make this method klass_name? so then you don't have to call set_klass_name then use klass_name. you can just use it